craft_parts/main.py

```diff
@@ -0,0 +1,304 @@
+        },
+        Step.PRIME: {
+            ActionType.RUN: "Prime",
+            ActionType.RERUN: "Re-prime",
+            ActionType.SKIP: "Skip prime",
+        },
+    }
+
+    message = f"{msg[action.step][action.action_type]} {action.part_name}"
+
+    if action.reason:
+        message += f" ({action.reason})"
+
+    return message
+
+
+def _parse_step(name: str) -> Step:
+    step_map = {
+        "pull": Step.PULL,
+        "build": Step.BUILD,
+        "stage": Step.STAGE,
+        "prime": Step.PRIME,
+    }
+
+    return step_map.get(name, Step.PRIME)
+
+
+def _parse_arguments() -> argparse.Namespace:
+    prog = "python -m craft_parts"
+    description = (
+        "A command line interface for the craft_parts module to build "
+        "parts-based projects."
+    )
+
+    parser = argparse.ArgumentParser(prog=prog, description=description, add_help=False)
+    parser.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        default=argparse.SUPPRESS,
+        help="Show this help message and exit.",
+    )
+    parser.add_argument(
+        "-f",
+        "--file",
+        metavar="filename",
+        default="parts.yaml",
+        help="The parts specification file. Default is 'parts.yaml'.",
+    )
+    parser.add_argument(
+        "--refresh",
+        action="store_true",
+        help="Update the stage packages list before procceeding.",
+    )
+    parser.add_argument(
+        "--plan-only",
+        action="store_true",
+        help="Show planned actions to be executed and exit.",
+    )
+    parser.add_argument(
+        "--show-skipped",
+        action="store_true",
+        help="Also display skipped actions.",
+    )
+    parser.add_argument(
+        "--work-dir",
+        metavar="dirname",
+        default=".",
+        help="Use the specified work directory. Defaults to current dir.",
+    )
+    parser.add_argument(
+        "--application-name",
+        metavar="name",
+        default="craft_parts",
+        help="Set the application name. Default is 'craft_parts'.",
+    )
+    parser.add_argument(
+        "--base",
+        metavar="name",
+        default="",
+        help="Use the specified build base. Defaults to host system.",
+    )
+    parser.add_argument(
+        "--cache-dir",
+        metavar="dirname",
+        default="",
+        help="Set an alternate cache directory location.",
+    )
+    parser.add_argument(
+        "--debug",
+        action="store_true",
+        help="Enable debug messages.",
+    )
+    parser.add_argument(
+        "--version",
+        action="store_true",
+        help="Display the craft-parts version and exit.",
+    )
+
+    help_parser = argparse.ArgumentParser(add_help=False)
+    help_parser.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        default=argparse.SUPPRESS,
+        help="Show this help message and exit.",
+    )
+
+    subparsers = parser.add_subparsers(dest="command")
```

Comment:
I would make this mandatory

```suggestion
    subparsers = parser.add_subparsers(dest="command", required=True)
```

Reply:
I like the idea, however that would generate an error for `python -mcraft_parts --version`.